qemu/tests/blockdev_commit_with_ioerror.py

```diff
@@ -0,0 +1,94 @@
+from aexpect import ShellTimeoutError
+
+from virttest import utils_numeric
+from virttest import utils_disk
+
+from provider import backup_utils
+from provider.blockdev_commit_base import BlockDevCommitTest
+
+
+class BlockdevCommitWithIoerror(BlockDevCommitTest):
+
+    def generate_tempfile(self, root_dir, filename, size="1000M", timeout=20):
+        """Generate temp data file in guest"""
+        self.session = self.main_vm.wait_for_login()
+        file_path = "%s/%s" % (root_dir, filename)
+        count = int(
+            utils_numeric.normalize_data_size(
+                size,
+                order_magnitude="M",
+                factor=1024))
+        dd_cmd = self.main_vm.params.get(
+            "dd_cmd", "dd if=/dev/urandom of=%s bs=1M count=%s oflag=direct")
+        mk_file_cmd = dd_cmd % (file_path, count)
+        try:
+            self.session.cmd(mk_file_cmd, timeout=timeout)
+        except ShellTimeoutError as e:
+            if self.params["error_msg"] not in str(e):
+                self.test.fail(str(e))
+            self.main_vm.verify_status("io-error")
+            self.files_info.append([root_dir, filename])
+        else:
+            self.test.fail("Can dd 1G file")
+
+    def md5_file(self):
+        if not self.session:
+            self.session = self.main_vm.wait_for_login()
+        output = self.session.cmd_output('\n', timeout=120)
+        if self.params["dd_done"] not in output:
+            self.test.fail("dd not continue to run after vm resume")
+        for info in self.files_info:
+            file_path = "%s/%s" % (info[0], info[1])
+            md5_cmd = "md5sum %s > %s.md5 && sync" % (file_path, file_path)
+            self.session.cmd(md5_cmd, timeout=120)
+        self.session.close()
```

Comment:
After dd done, we got a md5sum from the original file, after vm reboot, we got another. I think these two are always the same, as we've already covered that test. My point is, do we need to choose a original file to dd? I created a 1G file and got its md5sum before creating snapshot as the original file, then create snapshot, dd if=/the/1G/file of=xxx, we hit io-error, then do commit, dd done, we got another md5sum, compare them.
1G is too large, a 200M file is OK, for you create a snapshot in a 100M space